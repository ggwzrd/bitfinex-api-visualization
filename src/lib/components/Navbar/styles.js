
import get from "lodash/get"

const styles = theme => ({
    navigationWrapper: {
        zIndex: 2,
        transition: "all 300ms ease",
    },

    appbar: {
        backgroundColor: "transparent",
    },

    toolbar: {
        width: "100%",
        maxWidth: get(theme, "appicalTheme.sizes.pagesMaxWidth"),
        margin: "0 auto",
        padding: `0 ${get(theme, "appicalTheme.spacing.smallest")}`,
        justifyContent: "space-between",

        [theme.breakpoints.up("sm")]: {
            minHeight: get(theme, "appicalTheme.sizes.appBarHeight"),
        },
    },

    leftButton: {
        width: get(theme, "appicalTheme.sizes.iconButtonSize"),
        height: get(theme, "appicalTheme.sizes.iconButtonSize"),
        marginLeft: "-12px",
        marginRight: "20px",
    },

    title: {
        flex: "1",
        overflow: "hidden",
        whiteSpace: "nowrap",
        textOverflow: "ellipsis",
    },

    flex: {
        flex: "1",
    },

    transparent: {
        backgroundColor: "transparent",
        color: "inherit",
    },

    tabs: {
        paddingLeft: get(theme, "appicalTheme.spacing.largest"),

        [theme.breakpoints.down("xs")]: {
            paddingLeft: 0,
        },
    },
})

export default styles
