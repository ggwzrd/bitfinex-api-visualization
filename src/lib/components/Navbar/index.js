import React from "react"
import PropTypes from "prop-types"
import { Link } from "react-router-dom"
import Animated from "animated/lib/targets/react-dom"
import classNames from "classnames"

// material-ui
import AppBar from "@material-ui/core/AppBar/AppBar"
import Toolbar from "@material-ui/core/Toolbar/Toolbar"
import IconButton from "@material-ui/core/IconButton/IconButton"
import Typography from "@material-ui/core/Typography/Typography"
import withStyles from "@material-ui/core/styles/withStyles"

// icons
import { MdArrowBack } from "react-icons/md"

import styles from "./styles"

export const Navbar = ({
    color, classes, transparent, leftIcon, backLink, backFunc, title, rightElement, children, ...rest
}) => (
    <Animated.div className={classes.navigationWrapper}>
        <AppBar
            position="fixed"
            color={color}
            elevation={transparent ? 0 : 2}
            className={classNames(classes.appbar, {
                [classes.transparent]: transparent,
            })}
            {...rest}
        >
            <Toolbar className={classes.toolbar}>
                {backLink && (
                    <IconButton color="inherit" className={classes.leftButton} component={Link} to={backLink}>
                        {leftIcon || <MdArrowBack />}
                    </IconButton>
                )}
                {backFunc && (
                    <IconButton color="inherit" className={classes.leftButton} onClick={backFunc}>
                        {leftIcon || <MdArrowBack />}
                    </IconButton>
                )}
                <Typography type="title" color="inherit" className={classes.title}>
                    {title}
                </Typography>
                {children}
                {rightElement}
            </Toolbar>
        </AppBar>
    </Animated.div>
)

Navbar.propTypes = {
    title: PropTypes.string,
    backLink: PropTypes.string,
    backFunc: PropTypes.func,
    rightElement: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.element,
    ]),
    classes: PropTypes.object,
    transparent: PropTypes.bool,
    color: PropTypes.oneOf([
        "primary",
        "secondary",
        "inherit",
        "default",
    ]),
    leftIcon: PropTypes.node,
    children: PropTypes.node,
}

Navbar.defaultProps = {
    title: null,
    backLink: null,
    backFunc: null,
    rightElement: null,
    leftIcon: null,
    children: null,
    transparent: false,
    color: "primary",
    classes: {},
}

export default withStyles(styles)(Navbar)
