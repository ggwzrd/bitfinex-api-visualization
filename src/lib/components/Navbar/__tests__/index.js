import React from "react"
import createShallow from "@material-ui/core/test-utils/createShallow"
import Navbar from "../"

describe("<Navbar />", () => {
    it("Does not explode", () => {
        const component = createShallow()(<Navbar />)
        expect(component).toMatchSnapshot()
    })
})
