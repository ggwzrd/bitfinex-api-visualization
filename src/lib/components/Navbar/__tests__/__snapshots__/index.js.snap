// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Does not explode 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Navbar) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "backFunc": null,
      "backLink": null,
      "children": null,
      "classes": Object {
        "appbar": "Navbar-appbar-2",
        "flex": "Navbar-flex-6",
        "leftButton": "Navbar-leftButton-4",
        "navigationWrapper": "Navbar-navigationWrapper-1",
        "tabs": "Navbar-tabs-8",
        "title": "Navbar-title-5",
        "toolbar": "Navbar-toolbar-3",
        "transparent": "Navbar-transparent-7",
      },
      "color": "primary",
      "leftIcon": null,
      "rightElement": null,
      "title": null,
      "transparent": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backFunc": null,
        "backLink": null,
        "children": null,
        "classes": Object {
          "appbar": "Navbar-appbar-2",
          "flex": "Navbar-flex-6",
          "leftButton": "Navbar-leftButton-4",
          "navigationWrapper": "Navbar-navigationWrapper-1",
          "tabs": "Navbar-tabs-8",
          "title": "Navbar-title-5",
          "toolbar": "Navbar-toolbar-3",
          "transparent": "Navbar-transparent-7",
        },
        "color": "primary",
        "leftIcon": null,
        "rightElement": null,
        "title": null,
        "transparent": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
